File: solution/199/solution.cpp
#include <stdio.h>
#include <queue>
#include <vector>
struct TreeNode {
      int val;
      TreeNode *left;
      TreeNode *right;
      TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
class Solution {
public:
    std::vector<int> rightSideView(TreeNode* root) {
        std::queue<std::pair<TreeNode*, int> > nodequeue;
        std::vector<int> result;
        if(root)nodequeue.push(std::make_pair(root, 0));
        while(!nodequeue.empty()){
            TreeNode* node = nodequeue.front().first;
            int deep = nodequeue.front().second;
            nodequeue.pop();
            if(result.size()==deep)result.push_back(node->val);
            else result[deep] = node->val;
            if(node->left)nodequeue.push(std::make_pair(node->left, deep+1));
            if(node->right)nodequeue.push(std::make_pair(node->right, deep+1));
        }
        return result;
    }
};

File: solution/199/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        result = []
        queue = []
        if root:
            queue.append((root, 0))
        while len(queue) > 0:
            node = queue[0][0]
            deep = queue[0][1]
            del queue[0]
            if len(result) == deep:
                result.append(node.val)
            else:
                result[deep] = node.val
            if node.left:
                queue.append((node.left, deep+1))
            if node.right:
                queue.append((node.right, deep+1))
        return result